From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ivan Pekov <ivan@mrivanplays.com>
Date: Tue, 18 Aug 2020 11:00:26 +0300
Subject: [PATCH] Update distance map if necessary

I've seen people report to paper bad timings with the distance map lagging whilst
they have animals & monsters disabled in those specific worlds. This makes updating
the distance map only if animals or monsters are allowed to spawn in that world.

diff --git a/src/main/java/net/minecraft/server/ChunkProviderServer.java b/src/main/java/net/minecraft/server/ChunkProviderServer.java
index 5af513cb052382c34e157110265f80a1a2efef15..d50cb0a8b14ca0a57fcc1e6ee33eb7a2b67fbf6d 100644
--- a/src/main/java/net/minecraft/server/ChunkProviderServer.java
+++ b/src/main/java/net/minecraft/server/ChunkProviderServer.java
@@ -990,7 +990,7 @@ public class ChunkProviderServer extends IChunkProvider {
             int l = this.chunkMapDistance.b();
             // Paper start - per player mob spawning
             SpawnerCreature.d spawnercreature_d; // moved down
-            if (this.playerChunkMap.playerMobDistanceMap != null) {
+            if ((allowAnimals || allowMonsters) && this.playerChunkMap.playerMobDistanceMap != null) { // Yatopia
                 // update distance map
                 this.world.timings.playerMobDistanceMapUpdate.startTiming();
                 this.playerChunkMap.playerMobDistanceMap.update(this.world.players, this.playerChunkMap.viewDistance);
